data/src/main/java/com/microsoft/azure/kusto/data/KustoType.java

```diff
@@ -0,0 +1,55 @@
+package com.microsoft.azure.kusto.data;
+
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.util.UUID;
+import java.util.function.Function;
+
+/**
+ * Represents various ADX data types and their bindings to Java classes
+ * 
+ * @author alex
+ *
+ * @see https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/scalar-data-types/
+ *
+ * @param <C>
+ */
+public class KustoType<C> {
+	
+	public static final KustoType<Integer>			INTEGER						= new KustoType<>(Integer.class);
+	// Longs often return as "Integer", so we need to cast to "number", and then get the long value:
+	public static final KustoType<Long>				LONG						= new KustoType<>(Long.class,
+			o -> Long.valueOf(Number.class.cast(o).longValue()));
+	public static final KustoType<UUID>				GUID_UUID					= new KustoType<>(UUID.class, o -> UUID.fromString(o.toString()));
+	public static final KustoType<String>			GUID_STRING					= new KustoType<>(String.class);
+	public static final KustoType<String>			STRING						= new KustoType<>(String.class);
+	public static final KustoType<Double>			REAL_DOUBLE					= new KustoType<>(Double.class);
+	public static final KustoType<Float>			REAL_FLOAT					= new KustoType<>(Float.class,
+			o -> Float.valueOf(Double.class.cast(o).floatValue()));
+	
+	public static final KustoType<ZonedDateTime>	DATETIME_ZONED_DATE_TIME	= new KustoType<>(ZonedDateTime.class, o -> ZonedDateTime.parse(o.toString()));
+	public static final KustoType<Instant>			DATETIME_INSTANT			= new KustoType<>(Instant.class, o -> Instant.parse(o.toString()));
```

Comment:
Please add the following:
Duration and call it Timespan
Decimal 
Dynamic 
Boolean

Reply:
Sure, but since I don't have tables with these column types, could you tell me what Java types they currently return so I can write the conversion code?

Reply:
You should check it anyway but 
Timespan: String
decimal: String
dynamic: string or JSONObject 
boolean: Boolean

Reply:
These types make no sense to me. A timespan as a `String`? A decimal as a `String`?

Also, returning JSONObject is a terrible idea in my opinion since that's the type of the underlying JSON library. Why would you want to expose it? That would make user code depend on an internal dependency of yours, so you'll be stuck with it forever.

Reply:
I think you missunderstood me - i answered as to which types are iferered by the JSON library from these kusto types ... 
Kusto returns Timespan decimals and dynamic as strings because it is the most convenient to consume it by anyone calling the http query.
We should convert them like that:
Timespan: String -> Duration
decimal: String -> BigDecimal
dynamic : STring /JSONObject -> JSONObject / JSONArray (dynamic type is representing a .NET json object)
boolean : boolean -> boolean

Reply:
In that case, the timestamp, decimal and boolean make sense.

However, why would you like to expose `org.json.JSONObject ` directly to users? From my point of view, it's an implementation detail. If you ever decide to switch to some other JSON parsing library you'd be breaking user's code.